### PR TITLE
refactor: clean up deprecated content

### PR DESCRIPTION
**Pull request by xiyaowong**
Should we also remove the composite escape related content? Although it was added recently, I believe that most users have already completed the migration.

### CODE REVIEW

**Comment by theol0403** [comment]
Wait 1-2 months

**Comment by xiyaowong** [comment]
Then I merge it first?

**Comment by theol0403** [comment]
What do you mean?

**Comment by xiyaowong** [comment]
Merge this PR

**Comment by theol0403** [comment]
No, we should wait 1-2 months before merging this PR?

**Comment by xiyaowong** [comment]
The deleted content has been outdated for over half a year.

File: README.md [hunk 1]
```diff
@@ -591,13 +591,7 @@ Parameters:
...
 ### VimScript
 
-> **Note:** Since 1.0.0, vimscript functions are deprecated. Use the [Lua](#%EF%B8%8F-api) api instead.
-
--   `VSCodeNotify()`/`VSCodeCall()`: deprecated, use [Lua](#%EF%B8%8F-api) `require('vscode').call()` instead.
--   `VSCodeNotifyRange()`/`VSCodeCallRange()`: deprecated, use [Lua](#%EF%B8%8F-api)
-    `require('vscode').call(…, {range:…})` instead.
--   `VSCodeNotifyRangePos()`/`VSCodeCallRangePos()`: deprecated, use [Lua](#%EF%B8%8F-api)
-    `require('vscode').call(…, {range:…})` instead.
+You can use `v:lua.require("vscode")` to access the API from VimScript.
```
Comment (justinmk) [inline README.md:594]: The new text removed the link to the Lua text. And now ctrl-f on the readme won't find "VSCodeNotify".

I'm always in favor of removing tech debt, but this section is very small and minimal, thus low cost. It takes a long time for old blog posts, etc, to fade away. And we haven't removed the actual code. So this readme change is a regression.

File: package.json [hunk 2]
```diff
@@ -39,12 +39,6 @@
         "configuration": {
             "title": "Neovim",
             "properties": {
-                "vscode-neovim.neovimPath": {
```
Comment (justinmk) [inline package.json:42]: 👍 it's definitely time to remove these old settings

File: README.md (hunk 1 above)
Comment (xiyaowong) [reply README.md:594]: makes sense

**Review by theol0403 — APPROVED** [review]

**Review by justinmk — APPROVED** [review]

**Pull request merged**